### DISCUSSION TITLE
Please sort search params of the __manifest requests

### DISCUSSION THREAD

**Discussion by lilouartz** [remix-run/remix#9885] [Proposals]
We use a HTTP server that forces URL search parameter sorting.

What this means is that if URL is received with search parameters that are not sorted alphabetically, it will perform a redirect to a URL with the same search parameters sorted alphabetically, e.g. https://pillser.com/?b=x&a=x will redirect to https://pillser.com/?a=x&b=x.

This is causing an issue with manifest URLs because it is resulting in redirect, e.g.

https://pillser.com/__manifest?version=dc0f10e5&p=%2Fbest-price-guarantee&p=%2F&p=%2Fsign-up&p=%2Fsign-in&p=%2Fresearch-feed&p=%2Fbrands&p=%2Fprobiotics&p=%2Fvitamins&p=%2Fminerals&p=%2Fask&p=%2Fsearch&p=%2Fmedical-disclaimer&p=%2Faffiliate-disclosure&p=%2Fterms-of-service&p=%2Fprivacy-policy

will redirect to

https://pillser.com/__manifest?p=%2Fbest-price-guarantee&p=%2F&p=%2Fsign-up&p=%2Fsign-in&p=%2Fresearch-feed&p=%2Fbrands&p=%2Fprobiotics&p=%2Fvitamins&p=%2Fminerals&p=%2Fask&p=%2Fsearch&p=%2Fmedical-disclaimer&p=%2Faffiliate-disclosure&p=%2Fterms-of-service&p=%2Fprivacy-policy&version=dc0f10e5

It would be ideal if Remix sorts these parameters.

**Comment by GabenGar** [comment]
Define "alphabetically". [`URLSearchParams.sort()`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/sort) uses `unicode code points of the keys` which is kinda "alphabetically" but kinda not. Chances are your HTTP server has a slightly different (but still cache-breaking) opinion in this regard.
You should instead add `/__manifest` (or all bundle output files really) into exclusion from this rule. There is generally no reason to "prettify" search params not visible to the end user.
